crates/ruff_linter/src/rules/flake8_pyi/snapshots/ruff_linter__rules__flake8_pyi__tests__PYI055_PYI055.pyi.snap

```diff
@@ -121,11 +121,52 @@ PYI055.pyi:9:10: PYI055 [*] Multiple `type` members in a union. Combine them int
 7  7  | v: Union[type[float], type[complex]]
 8  8  | w: Union[type[float, int], type[complex]]
 9     |-x: Union[Union[type[float, int], type[complex]]]
-   9  |+x: Union[type[Union[float, int, complex]]]
+   9  |+x: type[Union[float, int, complex]]
 10 10 | y: Union[Union[Union[type[float, int], type[complex]]]]
 11 11 | z: Union[type[complex], Union[Union[type[float, int]]]]
 12 12 | 
 
+PYI055.pyi:10:4: PYI055 [*] Multiple `type` members in a union. Combine them into one, e.g., `type[Union[float, int, complex]]`.
+   |
+ 8 | w: Union[type[float, int], type[complex]]
+ 9 | x: Union[Union[type[float, int], type[complex]]]
+10 | y: Union[Union[Union[type[float, int], type[complex]]]]
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PYI055
+11 | z: Union[type[complex], Union[Union[type[float, int]]]]
+   |
+   = help: Combine multiple `type` members
+
+ℹ Safe fix
+7  7  | v: Union[type[float], type[complex]]
+8  8  | w: Union[type[float, int], type[complex]]
+9  9  | x: Union[Union[type[float, int], type[complex]]]
+10    |-y: Union[Union[Union[type[float, int], type[complex]]]]
+   10 |+y: type[Union[float, int, complex]]
```

Comment:
hmm, this isn't an equivalent rewrite. `type[float, int]` isn't a valid type expression (it would need to be either `type[float | int]` or `type[Union[float, int]]`, and therefore `Union[type[float, int], type[complex]]` isn't a valid type expression, and therefore `Union[Union[Union[type[float, int], type[complex]]]]` isn't a valid type expression. The rewrite here turns it into something that _is_ valid.

I think it would be better if we refused the temptation to guess here and did not emit the diagnostic at all if `type` is subscripted with multiple arguments.

Reply:
Makes sense